web/static/js/thesis-editor.js

```diff
@@ -16,35 +16,53 @@ import RawHtmlEditor from './content_types/raw_html_editor'
 import ImageEditor from './content_types/image_editor'
```

Comment:
My React-fu is still not the greatest, and there are parts here where MobX or Redux would be very useful. I'm thinking about the refactoring in #39. I think I'd have loads of parts and pieces everywhere during the refactor, though, so I'm not quite ready to tackle that yet.